src/Microsoft.AspNetCore.Mvc.Core/Internal/FileResultExecutorBase.cs

```diff
@@ -57,15 +57,17 @@ internal enum PreconditionState
             SetContentType(context, result);
             SetContentDispositionHeader(context, result);
             Logger.FileResultExecuting(result.FileDownloadName);
-            if (fileLength.HasValue)
-            {
-                SetAcceptRangeHeader(context);
-            }
 
             var request = context.HttpContext.Request;
             var httpRequestHeaders = request.GetTypedHeaders();
             var response = context.HttpContext.Response;
             var httpResponseHeaders = response.GetTypedHeaders();
+            if (fileLength.HasValue)
+            {
+                SetAcceptRangeHeader(context);
+                response.ContentLength = fileLength.Value;
```

Comment:
So is the idea here to set the content length assuming that it's not a 'range' request, and then overwrite the content length later if it is? I noticed there's some other code in this class that sets content length.

This deserves a comment if that's the case.